src/fidesops/task/graph_task.py

```diff
@@ -417,18 +445,11 @@ def filter_data_categories(
         if not target_field_paths:
             continue
 
-        # Normalize nested data into a flat dataframe
-        df: pd.DataFrame = pd.json_normalize(results, sep=".")
-        # Only keep intersection of dataframe columns and target field paths
-        df = df[
-            df.columns & [field_path.string_path for field_path in target_field_paths]
-        ]
-        # Turn the filtered results back into a list of dictionaries
-        filtered_flattened_results: List[Dict[str, Optional[Any]]] = df.to_dict(
-            orient="records"
-        )
-        for row in filtered_flattened_results:
-            # For each row, unflatten the dictionary
-            filtered_access_results[node_address].append(unflatten_dict(row))
+        for row in results:
+            filtered_results: Dict[str, Any] = {}
+            for field_path in target_field_paths:
+                select_and_save_field(filtered_results, row, field_path)
+            remove_empty_objects(filtered_results)
+            filtered_access_results[node_address].append(filtered_results)
```

Comment:
This is another primary change, `json_normalize` doesn't do what we need for array data, so I added a recursive `select_and_save_field` method that incrementally builds a new result with just the matched data categories on scalar, dictionary fields, array fields, or arrays of objects.